monitor/flowgen/flowgen.go

```diff
@@ -2,25 +2,185 @@ package flowgen
 
 import (
 	"context"
+	"fmt"
+	"math/big"
 	"time"
 
-	"github.com/omni-network/omni/lib/tokens"
+	"github.com/omni-network/omni/e2e/app/eoa"
+	"github.com/omni-network/omni/lib/contracts"
+	"github.com/omni-network/omni/lib/contracts/solvernet"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/ethclient/ethbackend"
+	"github.com/omni-network/omni/lib/evmchain"
+	"github.com/omni-network/omni/lib/log"
+	"github.com/omni-network/omni/lib/netconf"
+	"github.com/omni-network/omni/lib/xchain"
+	"github.com/omni-network/omni/monitor/flowgen/bridging"
+	"github.com/omni-network/omni/monitor/flowgen/types"
+	"github.com/omni-network/omni/monitor/flowgen/util"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
 )
 
-func Start(ctx context.Context) error {
-	jobs := []Job{
-		{
-			Name:    "Symbiotic",
-			Run:     newSymbioticRunner(),
-			Cadence: 1 * time.Hour,
-			Spend:   spend(tokens.WSTETH, ether1),
-		},
+func Start(
+	ctx context.Context,
+	xprovider xchain.Provider,
+	network netconf.Network,
+	rpcEndpoints xchain.RPCEndpoints,
+	keyPath string,
+) error {
+	var jobs []*types.Job
+
+	// Add jobs according to the current network ID
+	switch network.ID {
+	case netconf.Devnet:
+		// Instantiate bridging of native ETH on a devnet
+		job, err := bridging.NewJob(
+			netconf.Devnet,
+			evmchain.IDMockL1,
+			evmchain.IDMockL2,
+			eoa.RoleMonitor,
+			common.Address{}, // native ETH
+			big.NewInt(0).Mul(util.MilliEther, big.NewInt(2)), // 0.002 ETH
+		)
+		if err != nil {
+			return err
+		}
+
+		jobs = append(jobs, job)
+
+	default:
+	}
+
+	if keyPath == "" {
+		return errors.New("private key is required")
+	}
+
+	privKey, err := ethcrypto.LoadECDSA(keyPath)
+	if err != nil {
+		return errors.Wrap(err, "load key", "path", keyPath)
+	}
+
+	backends, err := ethbackend.BackendsFromNetwork(network, rpcEndpoints, privKey)
+	if err != nil {
+		return errors.Wrap(err, "backends")
 	}
 
-	_ = jobs
-	_ = ctx
+	for _, job := range jobs {
+		go func() {
+			ticker := time.NewTicker(job.Cadence)
+			defer ticker.Stop()
 
-	// TODO
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case <-ticker.C:
+					if err := run(ctx, xprovider, backends, job); err != nil {
+						log.Error(ctx, fmt.Sprintf("Flowgen job %s failed", job.Name), err)
```

Comment:
avoid fmt.Sprint in logging or errors, use structured erorrs. Also avoid `Error` level, rather change to `Warn` with`(will retry)`. See go style guide in notion for details.

Reply:
Stick to flowgen logging pattern: `log.Warn("Flowgen: job failed (will try)", err, "name", job.Name)`

Reply:
you can also add the job to the ctx for consistent logic: `jobCtx := log.WithCtx(ctx, "job", j.Name)`